WordPress/src/main/java/org/wordpress/android/ui/ActivityLauncher.java

```diff
@@ -600,27 +602,35 @@ public static void editPostOrPageForResult(Activity activity, SiteModel site, Po
             return;
         }
 
-        Intent intent = new Intent(activity, EditPostActivity.class);
+        Intent intent = EditPostBaseActivity.getNormalOrGutenbergEditPostActivityIntent(activity, false, post);
         intent.putExtra(WordPress.SITE, site);
         // PostModel objects can be quite large, since content field is not size restricted,
         // in order to avoid issues like TransactionTooLargeException it's better to pass the id of the post.
         // However, we still want to keep passing the SiteModel to avoid confusion around local & remote ids.
-        intent.putExtra(EditPostActivity.EXTRA_POST_LOCAL_ID, post.getId());
+        intent.putExtra(EditPostBaseActivity.EXTRA_POST_LOCAL_ID, post.getId());
         activity.startActivityForResult(intent, RequestCodes.EDIT_POST);
     }
 
     public static void editPageForResult(@NonNull Fragment fragment, @NonNull PageModel page) {
         Intent intent = new Intent(fragment.getContext(), EditPostActivity.class);
         intent.putExtra(WordPress.SITE, page.getSite());
-        intent.putExtra(EditPostActivity.EXTRA_POST_LOCAL_ID, page.getPageId());
+        intent.putExtra(EditPostBaseActivity.EXTRA_POST_LOCAL_ID, page.getPageId());
+        fragment.startActivityForResult(intent, RequestCodes.EDIT_POST);
+    }
+
+    public static void editPageForResultOnGutenberg(@NonNull Fragment fragment, @NonNull PageModel page) {
```

Comment:
Hmm, having a Gutenberg-dedicated method (`editPageForResultOnGutenberg`) is generally fine but, only a few lines above (line 605) we're seeing a different pattern, where the method is just one and it it decides internally which editor to launch. Can we get away with using just one of the two patterns perhaps?

Reply:
I think it's fine for this case as otherwise we need to inject and use FluxC in ActivityLauncher just to be able to inflate the PageModel, read the content of the Post body and then only decide which Activity to launch. This, in the only case where this method is called, is already known at the time the call happens by the caller.